builder/azure/arm/config.go

```diff
@@ -774,6 +781,23 @@ func assertTagProperties(c *Config, errs *packer.MultiError) {
 func assertRequiredParametersSet(c *Config, errs *packer.MultiError) {
 	c.ClientConfig.Validate(errs)
 
+	/////////////////////////////////////////////
+	// Identity
+	if len(c.UserAssignedManagedIdentities) != 0 {
+		for _, rid := range c.UserAssignedManagedIdentities {
+			r, err := client.ParseResourceID(rid)
+			if err != nil {
+				errs = packer.MultiErrorAppend(errs, err)
+			}
+			if r.Provider != "" && !strings.EqualFold(r.Provider, "Microsoft.ManagedIdentity") {
+				errs = packer.MultiErrorAppend(errs, fmt.Errorf("A valid user assigned managed identity resource id must have a correct resource provider"))
+			}
+			if r.ResourceType.String() != "" && !strings.EqualFold(r.ResourceType.String(), "userAssignedIdentities") {
```

Comment:
`r.ResourceType.String()` never returns an empty string, right? (so why do we check for that?)

Reply:
Same reason as above - to stop further checking if the previous errors fails.

Reply:
fixed now.